pkg/nodediscovery/nodediscovery.go

```diff
@@ -525,6 +528,60 @@ func (n *NodeDiscovery) mutateNodeResource(nodeResource *ciliumv2.CiliumNode) er
 				nodeResource.Spec.Azure.InterfaceName = c.Azure.InterfaceName
 			}
 		}
+
+	case ipamOption.IPAMAlibabaCloud:
+		nodeResource.Spec.AlibabaCloud = alibabaCloudTypes.Spec{}
+
+		instanceID, err := alibabaCloudMetadata.GetInstanceID(context.TODO())
+		if err != nil {
+			log.WithError(err).Fatal("Unable to retrieve InstanceID of own ECS instance")
+		}
```

Comment:
Are you aware of `instanceID` ever coming back empty in Alibaba? We've seen this behavior in AWS ENI mode, which resulted in a bug where the CiliumNode resource was corrupted due to an empty `instanceID`, causing the code to attach all ENIs into the resource. This caused pods to be assigned to ENIs that are not actually attached to the node itself. The symptom users saw was that these pods were unable to be have connectivity. See https://github.com/cilium/cilium/pull/15012.

Reply:
I don't expect to run into this ,ECS API is responsible provide correct data. For precaution i will add the check.

Reply:
The check for empty `instanceID` is still missing

Reply:
fixed